src/client/Client.cc

```diff
@@ -14137,9 +14137,9 @@ int Client::_rename(Inode *fromdir, const char *fromname, Inode *todir, const ch
   }
   if (cct->_conf.get_val<bool>("client_quota") && fromdir != todir) {
     Inode *fromdir_root =
-      fromdir->quota.is_enabled(QUOTA_MAX_FILES) ? fromdir : get_quota_root(fromdir, perm, QUOTA_MAX_FILES);
+      fromdir->quota.is_enabled(QUOTA_ANY) ? fromdir : get_quota_root(fromdir, perm, QUOTA_ANY);
```

Comment:
Oh, so the quota realm for both dirs were the root realm and that allowed the rename :/

How was this test passing earlier?

Reply:
This is odd.

Reply:
Did you check how this line got set as MAX_FILES? Was it in the original quota code, or did somebody change it deliberately?

Reply:
Yes, without that it was hard to know if this modification is the right thing to do. The commit where this change was introduced is mentioned in my commit. In fact, that commit has one more similar change, I am working on that now.